### PR TITLE
Orthographic projection frontend

### PR DESCRIPTION
**Pull request by sashankaryal**
## What changes are proposed in this pull request?

Frontend for #2656 

Screenshot 2023-02-17 at 4 02 21 PM

### Notes to the reviewer
- Key files: app/packages/looker/src/worker/threeDLabelProcessor.ts, app/packages/looker/src/overlays/detection.ts

#### Example dataset
```python
import fiftyone as fo
import fiftyone.utils.utils3d as fou3d
import fiftyone.zoo as foz

quickstart_groups = foz.load_zoo_dataset(name="quickstart-groups", max_samples=12)
pcd_only_dataset = quickstart_groups.select_group_slices("pcd").clone()

pcd_only_dataset.delete_sample_field("group")

fou3d.compute_orthographic_projection_images(pcd_only_dataset, (-1,512),"/tmp/maps")

session = fo.launch_app(dataset=pcd_only_dataset)
```

#### Looker
- Add `Sample3D` modal
- Add looker for point clouds
- Add module to compute 3D detection boxes projections

#### Looker3D
- Adds support for dynamically adjusting point cloud size with infinite range

## How is this patch tested? If it is not, please explain why.

Locally.

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [x] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/looker-3d/src/action-bar/point-size.tsx [hunk 1]
```diff
@@ -1,50 +1,108 @@
 import { PopoutSectionTitle, useTheme } from "@fiftyone/components";
-import { Slider } from "@fiftyone/core/src/components/Common/RangeSlider";
+// import { Slider } from "@fiftyone/core/src/components/Common/RangeSlider";
```
Comment (benjaminpkane) [inline app/packages/looker-3d/src/action-bar/point-size.tsx:2]:
```suggestion
```

File: app/packages/looker-3d/src/action-bar/point-size.tsx [hunk 2]
```diff
@@ -1,50 +1,108 @@
...
 import * as recoil from "recoil";
 import { ActionItem } from "../containers";
 import {
   ACTION_SET_POINT_SIZE,
   currentActionAtom,
   currentPointSizeAtom,
-  pointSizeRangeAtom,
 } from "../state";
 import { ActionPopOver } from "./shared";
+import "./styles.css";
```
Comment (benjaminpkane) [inline app/packages/looker-3d/src/action-bar/point-size.tsx:15]: Please use CSS modules. For mui overrides, I think there are more formal ways to override styles

**Review by benjaminpkane — COMMENTED** [review]
The current codebase convention is to name react component files by their default export (capitalized camel case).

I've only briefly skimmed here, but I think there is an opportunity here to add some tests for the new code paths

File: app/packages/looker/src/worker/threed-label-processor.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      const [lx, ly] = label.dimensions as DetectionLabel["dimensions"];
+
+      minX = Math.min(minX, x - lx / 2);
+      maxX = Math.max(maxX, x + lx / 2);
+      minY = Math.min(minY, y - ly / 2);
+      maxY = Math.max(maxY, y + ly / 2);
+    }
+  }
+
+  const padding = 100;
```
Comment (manivoxel51) [inline app/packages/looker/src/worker/threed-label-processor.ts:99]: nit: move to constant? I have little context but will these numbers be configurable by users in the future?

File: app/packages/looker/src/worker/threed-label-processor.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+  inferredParamsCache[sample.id] = {
+    width: minX === Infinity ? 512 : maxX - minX + padding,
+    height: minY === Infinity ? 512 : maxY - minY + padding,
+    min_bound: [
+      minX === Infinity ? -100 : minX - padding,
```
Comment (manivoxel51) [inline app/packages/looker/src/worker/threed-label-processor.ts:105]: could you use `padding` here

**Review by manivoxel51 — COMMENTED** [review]
nice work Sashank! 🍨 

I don't have much context on prior work but did check out the two main files you mentioned in the PR.

**Review by benjaminpkane — APPROVED** [review]
LGTM!

**Pull request merged**